### PR TITLE
Dashboard favorites

### PR DESCRIPTION
**Pull request by SARodrigues**
## Description

**In this PR**  

- `BadgeNavigation` changes:  
  _The new `DashboardFavoritesLayout` uses this component as its sidebar's navigation_
  - Added a new `simple` theme  
  - Added the `badgePosition` prop so we can choose the position of the (number) badges  
  - Added the `orientation` prop, so we can display the navigation both horizontally and vertically  
- Added a `CardHoverToDelete` component: 
  _A component we can use to wrap `ProfileCard` and `ProjectCard` components in, in order to display the "Delete" icon when hovering the card._
- `account/favoritesService` service:  
  - Added services & hooks to retrieve the list of favorited: 
    - Projects
    - Project developers  
    - Investors 
- Other services:  
  - Favorite projects, project developers, investors query invalidation  
    _When the user favorites/un-favorites projects, project developers, investors_
- `Enums` changes: 
  - Added `Paths` for the dashboard favorites' pages 
  - Added `Queries` for the queries used to fetch the lists of favorited projects, project developers, and  investors.
- `DashboardLayout` changes:  
  - Added `sidebar` support  
    _Needed for the dashboard/favorites views._
- Added `DashboardFavoritesLayout`  
  _This layout is mostly a wrapper around the `DashboardLayout`. It adds the sidebar and is responsible for fetching the list of favorited projects, project developers, ~open calls~ to display on the navigation badges. The logic is similar to how we fetch data in the `Discover` layout._ 
- Added `dashboard/favorites/*` pages  
  - Projects
    _List of favorited projects_
  - Project developers
    _List of favorited project developers_
  - Investors
    _List of favorited investors_
  - Open calls
    _Placeholder only, given we don't have open calls implemented yet._

**Notes:** 
  - `Pagination` has not been implemented. It'll be addressed in [LET-853](https://vizzuality.atlassian.net/browse/LET-853)
  - Removing Projects, Project developers and Investors from the favorites will be addressed in [LET-854](https://vizzuality.atlassian.net/browse/LET-854), [LET-855](https://vizzuality.atlassian.net/browse/LET-854), [LET-855](https://vizzuality.atlassian.net/browse/LET-856)

## Testing instructions

Verify that, in the Dashboard:  
  - The user can see a list of favorited Projects, Project developers, investors  
  - The lists are correct  
  - The open calls page displays a message stating the user doesn't have favorited open calls yet  
  - Hovering over a Project, Project developer or investor card shows the "delete" icon  
  - The navigation works correctly 

## Tracking

[LET-613 | FE - Adapt BadgeNavigation component to be able to also display items vertically](https://vizzuality.atlassian.net/browse/LET-613)
[LET-614 | FE - DashboardLayout to support sidebars  ](https://vizzuality.atlassian.net/browse/LET-614)
[LET-840 | FE - Add DashboardFavourites layout](https://vizzuality.atlassian.net/browse/LET-840)

[LET-615 | FE - Signed in PDs and Investors can see a list of favorited Projects](https://vizzuality.atlassian.net/browse/LET-615)
[LET-616 | FE - Signed in PDs and Investors can see a list of favorited Project developers](https://vizzuality.atlassian.net/browse/LET-616)
[LET-617 | FE - Signed in PDs and Investors can see a list of favorited Investors](https://vizzuality.atlassian.net/browse/LET-617)

## Screen recording 

**Before adding the delete buttons** 

https://user-images.githubusercontent.com/6273795/182606183-26d8b724-005f-458c-a69e-d28a9119c7fa.mp4

**With the Delete buttons**  

https://user-images.githubusercontent.com/6273795/182624829-b3fea743-96cc-4f06-8580-0cac219dede8.mp4

### CODE REVIEW

**Comment by SARodrigues** [comment]
@barbara-chaves I apologize about the size of the PR. It was easier to do it all at once since the layout/content is directly related, but I see now that quite a few tasks could have been PR'd separately (eg: Badge navigation changes).

I can split this into different PRs if you prefer and is easier for you to review.

**Comment by SARodrigues** [comment]
Moving to WIP due to a bug: 
Screenshot 2022-08-03 at 16 00 35

File: frontend/pages/dashboard/favorites/projects.tsx [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+    console.log('unfavorite', id);
+  };
+
+  const handleRemoveAllClick = () => {
+    console.log('unfavorite all projects');
+  };
+
+  return (
+    <>
+      <div className="top-0 flex justify-between px-2 pb-1 mb-4 -mx-2 md:pt-10 md:-mt-10 md:pl-3 md:pr-2 lg:z-20 lg:sticky bg-background-dark">
```
Comment (barbara-chaves) [inline frontend/pages/dashboard/favorites/projects.tsx:49]:
```suggestion
      <div className="top-0 flex justify-between px-2 pb-1 mb-4 -mx-2 md:pt-10 md:-mt-10 md:pl-3 md:pr-2 lg:sticky bg-background-dark">
```

File: frontend/layouts/dashboard-favorites/component.tsx [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+    projects,
+    isLoadingProjects,
+    isFetchingProjects,
+    projectDevelopers,
+    isLoadingProjectDevelopers,
+    isFetchingProjectDevelopers,
+    investors,
+    isLoadingInvestors,
+    isFetchingInvestors,
+  ]) || { data: [], meta: [] };
```
Comment (barbara-chaves) [inline frontend/layouts/dashboard-favorites/component.tsx:95]:
```suggestion
  const projects = useFavoriteProjectsList(defaultQueryParams, defaultQueryOptions);

  const investors = useFavoriteInvestorsList(defaultQueryParams, defaultQueryOptions);

  const projectDevelopers = useFavoriteProjectDevelopersList(
    defaultQueryParams,
    defaultQueryOptions
  );

  // Change for real open calls
  const openCalls = useMemo(
    () => ({
      data: [],
      meta: { total: 0 },
      loading: false,
    }),
    []
  );

  const stats = useMemo(
    () => ({
      projects: projects.data?.meta.total,
      projectDevelopers: projectDevelopers.data?.meta.total,
      investors: investors.data?.meta.total,
      openCalls: 0,
    }),
    [projects, investors, projectDevelopers]
  );

  const getFavoriteCurrentData = (data: UseQueryResult<PagedResponse<any>>) => {
    return {
      data: data.data?.data,
      meta: data.data?.meta,
      loading: data.isLoading || data?.isFetching,
    };
  };

  const { data, meta, loading } = useMemo(() => {
    switch (pathname) {
      case Paths.DashboardFavoritesProjects:
        return getFavoriteCurrentData(projects);
      case Paths.DashboardFavoritesProjectDevelopers:
        return getFavoriteCurrentData(projectDevelopers);
      case Paths.DashboardFavoritesInvestors:
        return getFavoriteCurrentData(investors);
      case Paths.DashboardFavoritesOpenCalls:
        // Change when real open calls are implemented
        // return getFavoriteCurrentData(openCalls);
        return openCalls;
    }
  }, [pathname, projects, projectDevelopers, investors, openCalls]) || { data: [], meta: [] };
```

Comment (barbara-chaves) [reply frontend/layouts/dashboard-favorites/component.tsx:95]: My suggestion

**Review by barbara-chaves — APPROVED** [review]
I'm approving. I leave a suggestion, but you can merge the way it is.

File: frontend/layouts/dashboard-favorites/component.tsx (hunk 2 above)
Comment (SARodrigues) [reply frontend/layouts/dashboard-favorites/component.tsx:95]: Thank you. Your solution is way cleaner!

**Pull request merged**